force-app/main/default/classes/SOQL.cls

```diff
@@ -315,18 +524,51 @@ public with sharing class SOQL {
     public static void cloneLineItems(Id origOppId, List<Id> cloneOppIds) {
         // Query the original Opportunity Line Items
 
+      List<OpportunityLineItem> originalOpportunityLineItems=  [SELECT Id, Product2Id, Quantity, UnitPrice, PricebookEntryId,
+      
+        Name, OpportunityId FROM OpportunityLineItem
+        
+        WHERE OpportunityId = :origOppId];
+
         // Create a list to hold the new line items
+      if(originalOpportunityLineItems.isEmpty()){
 
+        return;
+      }
+
+      List<OpportunityLineItem> clonedLineItems = new List<OpportunityLineItem>();
         // Generate a random number between 1 and 3
 
-        // Loop through each cloned Opportunity Id
+        Integer randomNumber = (Integer)(Math.floor(Math.random() * 3) + 1);
+
 
-            // Loop through each original line item
+        for(Id cloneOppId : cloneOppIds) {// Loop through each cloned Opportunity Id
+
+            for(OpportunityLineItem origItem : originalOpportunityLineItems) {// Loop through each original line item
 
                 // Clone the original line item and add the random number to the quantity
+                OpportunityLineItem newItem = new OpportunityLineItem();
+                newItem.OpportunityId = cloneOppId;
+                newItem.Product2Id = origItem.Product2Id;
+                newItem.Quantity = origItem.Quantity + (Integer)(Math.floor(Math.random() * 3) + 1); // Add random 1-3 to quantity
+                newItem.UnitPrice = origItem.UnitPrice;
+                newItem.PricebookEntryId = origItem.PricebookEntryId;
+
+                clonedLineItems.add(newItem);
```

Comment:
Nicely done! cloning individual fields.
Try to use the Clone() method and try to use the variable randomNumber for the quantity